File: src/test/java/org/goafabric/calleeservice/service/CalleeServiceAdapter.java
package org.goafabric.calleeservice.service;

import org.eclipse.microprofile.rest.client.inject.RegisterRestClient;

import javax.ws.rs.GET;
import javax.ws.rs.Path;
import javax.ws.rs.PathParam;
import javax.ws.rs.QueryParam;

@Path("/callees")
@RegisterRestClient(baseUri = "http://localhost:50900")
public interface CalleeServiceAdapter {
    @GET
    @Path("sayMyName")
    Callee sayMyName(@QueryParam("name") String name);

    @GET
    @Path("sayMyOtherName/{name}")
    Callee sayMyOtherName(@PathParam("name") String name);

    @GET
    @Path("setSleepTime")
    Callee setSleepTime(@QueryParam("sleepTime") Long sleepTime);
}

File: src/test/java/org/goafabric/calleeservice/service/CalleeServiceAdapter.kt
package org.goafabric.calleeservice.service

import org.eclipse.microprofile.rest.client.inject.RegisterRestClient
import javax.ws.rs.GET
import org.goafabric.calleeservice.service.Callee
import javax.ws.rs.Path
import javax.ws.rs.PathParam
import javax.ws.rs.QueryParam

@Path("/callees")
@RegisterRestClient
interface CalleeServiceAdapter {
    @GET
    @Path("sayMyName")
    fun sayMyName(@QueryParam("name") name: String): Callee

    @GET
    @Path("sayMyOtherName/{name}")
    fun sayMyOtherName(@PathParam("name") name: String): Callee

    @GET
    @Path("setSleepTime")
    fun setSleepTime(@QueryParam("sleepTime") sleepTime: Long): Callee
}